pkg/driver/node_test.go

```diff
@@ -0,0 +1,427 @@
+			Mount: &csi.VolumeCapability_MountVolume{},
+		},
+		AccessMode: &csi.VolumeCapability_AccessMode{
+			Mode: csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER,
+		},
+	}
+	testCases := []struct {
+		name           string
+		req            *csi.NodeStageVolumeRequest
+		expErrCode     codes.Code
+		fakeMountPoint *mount.MountPoint
+	}{
+		{
+			name: "success normal",
+			req: &csi.NodeStageVolumeRequest{
+				PublishContext:    map[string]string{"devicePath": "/dev/fake"},
+				StagingTargetPath: "/test/path",
+				VolumeCapability:  stdVolCap,
+				VolumeId:          "vol-test",
+			},
+		},
+		{
+			name: "fail no VolumeId",
+			req: &csi.NodeStageVolumeRequest{
+				PublishContext:    map[string]string{"devicePath": "/dev/fake"},
+				StagingTargetPath: "/test/path",
+				VolumeCapability:  stdVolCap,
+			},
+			expErrCode: codes.InvalidArgument,
+		},
+		{
+			name: "fail no StagingTargetPath",
+			req: &csi.NodeStageVolumeRequest{
+				PublishContext:   map[string]string{"devicePath": "/dev/fake"},
+				VolumeCapability: stdVolCap,
+				VolumeId:         "vol-test",
+			},
+			expErrCode: codes.InvalidArgument,
+		},
+		{
+			name: "fail no VolumeCapability",
+			req: &csi.NodeStageVolumeRequest{
+				PublishContext:    map[string]string{"devicePath": "/dev/fake"},
+				StagingTargetPath: "/test/path",
+				VolumeId:          "vol-test",
+			},
+			expErrCode: codes.InvalidArgument,
+		},
+		{
+			name: "fail no devicePath",
+			req: &csi.NodeStageVolumeRequest{
+				StagingTargetPath: "/test/path",
+				VolumeCapability:  stdVolCap,
+				VolumeId:          "vol-test",
+			},
+			expErrCode: codes.InvalidArgument,
+		},
+		{
+			// To test idempotency we need to test the
+			// volume corresponding to the volume_id is
+			// already staged to the staging_target_path
+			// and the Plugin replied with OK. To achieve
+			// this we setup the fake mounter to return
+			// that /dev/fake is mounted at /test/path.
+			name: "success idempotency",
+			req: &csi.NodeStageVolumeRequest{
+				PublishContext:    map[string]string{"devicePath": "/dev/fake"},
+				StagingTargetPath: "/test/path",
+				VolumeCapability:  stdVolCap,
+				VolumeId:          "vol-test",
+			},
+			fakeMountPoint: &mount.MountPoint{
+				Device: "/dev/fake",
+				Path:   "/test/path",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			fakeMounter := NewFakeMounter()
+			if tc.fakeMountPoint != nil {
+				fakeMounter.MountPoints = append(fakeMounter.MountPoints, *tc.fakeMountPoint)
+			}
+			awsDriver := NewFakeDriver("", fakeMounter)
+
+			_, err := awsDriver.NodeStageVolume(context.TODO(), tc.req)
+			if err != nil {
+				srvErr, ok := status.FromError(err)
+				if !ok {
+					t.Fatalf("Could not get error status code from error: %v", srvErr)
+				}
+				if srvErr.Code() != tc.expErrCode {
+					t.Fatalf("Expected error code %d, got %d message %s", tc.expErrCode, srvErr.Code(), srvErr.Message())
+				}
+				// test succeeded
```

Comment:
```suggestion
			} else {
				if tc.expErrCode != codes.OK {
					t.Fatalf("Expected error %v, got no error", tc.expErrCode)
				}
			}
```

This way you don't have to `return` from inside the test.

Reply:
Good catch! I'll change the other tests to do this as well.